File: CSC330/inClass/perl/Machine.pl
#!/usr/bin/perl
use strict;
use warnings;
use Drink; 

my @drinks = ( Drink->new("Coke"),  Drink->new("Sprite"),
               Drink->new("Guinness", 6.50),  Drink->new("Bud Light", 0.25));

for ( my $i=0; $i<scalar @drinks; $i++) {
   $drinks[$i]->refill();
}

for ( my $i=0; $i<8; $i++) {
   $drinks[0]->vend();	
   }

for ( my $i=0; $i<2; $i++) {
   $drinks[1]->vend();
   }

for ( my $i=0; $i<11; $i++) {
   $drinks[2]->vend();
   }

for ( my $i=0; $i<1; $i++) {
   $drinks[3]->vend();
   }

my $totalProfit = 0.0;
for ( my $i=0; $i<4; $i++) {
   $totalProfit += $drinks[$i]->profit();
   }
  
print "\n Total Profit from Drink Sales: \$" . $totalProfit . "\n\n";

for ( my $i=0; $i<4; $i++) {
   print "You sold " . $drinks[$i]->drinksSold() . " " . $drinks[$i]->getDrinkName() . " for a total of " .  $drinks[$i]->profit() . "\n" . "Drinks needed to restock: ". $drinks[$i]->restockAmount(). "\n"; 
}

File: CSC330/inClass/Vending/drinkJava/Machine.java
public class Machine {

   public static void main( String[] args) {

 Drink drink1 = new Drink("Coke");
    Drink drink2 = new Drink("Sprite");
    Drink drink3 = new Drink("Blue Powerade", 2.75);
    Drink drink4 = new Drink("Guinness",6.25);


    drink1.refill();
    drink2.refill();
    drink3.refill();
    drink4.refill();


drink1.vend();
    drink1.vend();
    drink1.vend();
    drink1.vend();
    drink1.vend();
    drink1.vend();
    drink1.vend();
    drink1.vend();

 drink2.vend();
    drink2.vend();
    drink2.vend();
    drink2.vend();
    drink2.vend();
    drink2.vend();

drink3.vend();
    drink3.vend();


 drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();
    drink4.vend();

int totalSold = drink1.drinksSold() + drink2.drinksSold() +
                    drink3.drinksSold() + drink4.drinksSold();

    double totalProfit = drink1.profit() + drink2.profit() +
                         drink3.profit() + drink4.profit();

    System.out.print("A Total of " + totalSold + " drinks were ");
    System.out.println("sold for a profit of $" + totalProfit);
    System.out.println();
    System.out.println( drink1.restockAmount() + " " + drink1.getDrinkName() + " needed to restock machine");
    System.out.println( drink2.restockAmount() + " " + drink2.getDrinkName() + " needed to restock machine");
    System.out.println( drink3.restockAmount() + " " + drink3.getDrinkName() + " needed to restock machine");
    System.out.println( drink4.restockAmount() + " " + drink4.getDrinkName() + " needed to restock machine");

      }
}
